e2e/tests/regression/test_volume.robot

```diff
@@ -65,3 +66,18 @@ Test RWX Volume Without Migratable Should Be Incompatible With Strict-Local
 
     When Create volume 0 with    numberOfReplicas=1    migratable=False    accessMode=RWX    dataLocality=strict-local    dataEngine=${DATA_ENGINE}
     Then No volume created
+
+Test Volume Attached at Maximum Snapshot Count
+    [Tags]    volume    snapshot-limit
+    [Documentation]    Validate that reaching the snapshot limit of 250 does not affect
+    ...                volume attach/detach operations.
+    ...                Issue: https://github.com/longhorn/longhorn/issues/10308
+    Given Create volume 0 with    dataEngine=${DATA_ENGINE}    numberOfReplicas=1
+    When Attach volume 0
+    And Wait for volume 0 healthy
+    FOR    ${i}    IN RANGE    249
+        Create snapshot ${i} of volume 0
+    END
```

Comment:
Suggest to wrap this in a keyword, e.g, `Create ${snapshot_count} snapshot for volume ${volume_id}`